capstone/src/main/java/com/google/sps/servlets/LessonHandlerServlet.java

```diff
@@ -64,15 +65,44 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     public List<Lesson> getLessons(Room room) {
         List<Lesson> lessons = new ArrayList<>();
+        List<Key> nullKeys = new ArrayList<>();
     
         try {
             for(Key key : room.getAllLessons()) {
-                lessons.add(DatabaseService.getLesson(key));
+                Boolean status = checkLessonStatus(key);
+
+                if (status) {
+                    lessons.add(DatabaseService.getLesson(key));
+                } else {
+                    nullKeys.add(key);
+                }
             }
         }
         catch (EntityNotFoundException e) {
             System.err.println("Lesson entities don't exist.");
         }
+        removeDeletedLessons(room, nullKeys);
         return lessons;
     }
+
+    // Checks to see if the lesson still exists in datastore
+    public Boolean checkLessonStatus(Key key) {
```

Comment:
You should basically never use the Boolean type since autoboxing will handle turning a primitive `boolean` into a Boolean when it needs to be cast to an object.